### ISSUE TITLE
Implementation of repeated sampling

### ISSUE THREAD

**Issue by alexandre-eymael** [minhoooo1/CatMAE#4]
Hello,

Thank you for your work.

I am not sure to understand how you implemented your repeated sampling. I understand that you multiply the size of your dataset by the repeated sampling factor:

https://github.com/minhoooo1/CatMAE/blob/b37d24b775289658420042c7f68b6c8412b6fb0b/kinetics_dataset.py#L62

and then divide the index by this factor while fetching new data:

https://github.com/minhoooo1/CatMAE/blob/b37d24b775289658420042c7f68b6c8412b6fb0b/kinetics_dataset.py#L89

I agree that this implementation would work if the data were not shuffled (i.e., if we indeed had repeated samples within the same batch), but it seems you are using a distributed sampler that shuffles the data (i.e., with `shuffle`=True):

https://github.com/minhoooo1/CatMAE/blob/b37d24b775289658420042c7f68b6c8412b6fb0b/main_pretrain.py#L193-L195

In my understanding, you are not using the same samples `repeated_sampling` times within the same batch, but are simply doubling the number of epochs, since repeated samplings can occur in different batches.

Can you please tell me where I am wrong?

Thank you!

**Comment by minhoooo1** [comment]
Yes, thank you for pointing out my mistake. All of my pre-training experiments were indeed conducted based on 'simply doubling the number of epochs', and the actual epochs were recorded. This does not affect the fairness when comparing with other methods.
